conan.cmake

```diff
@@ -197,19 +197,30 @@ endmacro()
 
 macro(conan_cmake_run)
     parse_arguments(${ARGV})
-    conan_cmake_generate_conanfile(${ARGV})
-    if(CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
+
+    # TODO: Better detection
+    if(CONAN_EXPORTED OR CONAN_COMPILER)
```

Comment:
Agree, this is a temporary workaround before CONAN_EXPORTED was defined by conan, but it's supposed to work with both 0.18 and 0.19.